ouroboros-consensus-cardano/tools/db-validator/Block/Byron.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Block.Byron (
+    Args (..)
+  , ByronBlockArgs
+  , openGenesisByron
+  ) where
+
+import           Control.Monad.Except
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import           GHC.Natural (Natural)
+
+import           Cardano.Binary (Raw, unAnnotated)
+import qualified Cardano.Crypto as Crypto
+
+import qualified Cardano.Chain.Block as Chain
+import qualified Cardano.Chain.Genesis as Genesis
+import qualified Cardano.Chain.Update as Update
+import qualified Cardano.Chain.UTxO as Chain
+
+import           Ouroboros.Consensus.Node.ProtocolInfo
+import           Ouroboros.Consensus.Storage.ChainDB.Serialisation (SizeInBytes)
+
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock)
+import qualified Ouroboros.Consensus.Byron.Ledger as Byron
+import           Ouroboros.Consensus.Byron.Node (PBftSignatureThreshold,
+                     protocolInfoByron)
+
+import           HasAnalysis
+
+instance HasAnalysis ByronBlock where
+    data Args ByronBlock =
+      ByronBlockArgs {
+          configFileByron      :: FilePath
+        , requiresNetworkMagic :: Bool
+        , genesisHash          :: Maybe (Crypto.Hash Raw)
+        , threshold            :: Maybe PBftSignatureThreshold
+        }
+    mkProtocolInfo ByronBlockArgs {..} = do
+      config <- openGenesisByron configFileByron genesisHash requiresNetworkMagic
+      return $ mkByronProtocolInfo config threshold
+    countTxOutputs = aBlockOrBoundary (const 0) countTxOutputsByron
+    blockHeaderSize = fromIntegral .
+      aBlockOrBoundary blockBoundaryHeaderSize blockHeaderSizeByron
+    blockTxSizes = aBlockOrBoundary (const []) blockTxSizesByron
+    knownEBBs = const Byron.knownEBBs
+
+type ByronBlockArgs = Args ByronBlock
```

Comment:
~Is this type used/needed? It saves just one character, that's not worth the extra indirection for me. It's not that we have to be careful to maintain backwards compatibility. Moreover, it's easier to align multiple lines of `Args x` than `xArgs` :slightly_smiling_face:.~

UPDATE: I just the similar synonyms for Shelley and Cardano, which do save a lot of characters. Then we should keep this one too, for consistency, of course. So you can ignore this comment.